fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/idformat/RecordIdFormat.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * RecordIdFormat.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.idformat;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RecordIdFormat {
+    public enum FormatElementType implements FormatElement {
+        NONE(0),
+        NULL(1),
+        INT32(5),
+        INT32_OR_NULL(5),
+        INT64(9),
+        INT64_OR_NULL(9),
+        UUID_AS_STRING(18),
```

Comment:
Isn't this `17`? It doesn't matter much because it's rounded up to a multiple of `9`, but isn't this the longest Tuple-encoding that the type can have?

Reply:
Yes, that's right.